src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateClaimCallbackHandler.java

```diff
@@ -123,8 +131,23 @@ public List<CaseEvent> handledEvents() {
     }
 
     private CallbackResponse startClaim(CallbackParams callbackParams) {
+        List<LocationRefData> locations = fetchLocationData(callbackParams);
+
+        CaseData caseData = CaseData.builder()
+            .claimStarted(YES)
+            .courtLocation(CourtLocation.builder()
+                               .applicantPreferredCourtLocationList(courtLocationUtils.getLocationsFromList(locations))
+                               .build())
+            .build();
+
         return AboutToStartOrSubmitCallbackResponse.builder()
-            .data(CaseData.builder().claimStarted(YES).build().toMap(objectMapper)).build();
+            .data(caseData.toMap(objectMapper)).build();
+    }
+
+    private List<LocationRefData> fetchLocationData(CallbackParams callbackParams) {
+        String authToken = callbackParams.getParams().get(BEARER_TOKEN).toString();
+        // Potentially renaming this method to something more generic not DJ
```

Comment:
think you left a task in here, needs deleting whether youre gunna rename or not

Reply:
removed it!